posthog/api/instance_settings.py

```diff
@@ -49,20 +49,30 @@ def get_instance_setting(key: str, setting_config: Dict = {}) -> InstanceSetting
 
 
 class InstanceSettingsSerializer(serializers.Serializer):
-    key = serializers.CharField()
+    key = serializers.CharField(read_only=True)
     value = serializers.JSONField()  # value can be bool, int, or str
     value_type = serializers.CharField(read_only=True)
     description = serializers.CharField(read_only=True)
-    editable = serializers.BooleanField()
+    editable = serializers.BooleanField(read_only=True)
 
     def update(self, instance: InstanceSetting, validated_data: Dict[str, Any]) -> InstanceSetting:
         if instance.key not in SETTINGS_ALLOWING_API_OVERRIDE:
             raise serializers.ValidationError("This setting cannot be updated from the API.", code="no_api_override")
-        if validated_data["value"]:
-            target_type = settings.CONFIG[instance.key][2]
-            new_value_parsed = cast_str_to_desired_type(validated_data["value"], target_type)
-            setattr(config, instance.key, new_value_parsed)
-            instance.value = new_value_parsed
+
+        if not validated_data["value"]:
+            raise serializers.ValidationError({"value": "This field is required."}, code="required")
+
+        target_type = settings.CONFIG[instance.key][2]
+        new_value_parsed = cast_str_to_desired_type(validated_data["value"], target_type)
+
+        if instance.key == "RECORDINGS_TTL_WEEKS":
+            from ee.clickhouse.models.session_recording_event import update_recordings_ttl
+
+            update_recordings_ttl(new_value_parsed)
```

Comment:
Not sure we can run this query sync, but it'd be great cause if something goes wrong the config value won't get updated and the user will get an error.